sympy/utilities/misc.py

```diff
@@ -355,7 +355,8 @@ def translate(s, a, b=None, c=None):
 
     mr = {}
     if a is None:
-        assert c is None
+        if c is not None:
+            raise TypeError('Called translate with 4 arguments and a second argument of None, expected 3 arguments if second argument is None')
```

Comment:
Fixed.  I'm aware of PEP8, and I'm not going to argue about good programming practice, but some projects use longer line lengths or choose to exempt error messages and some other types of strings from the line length requirement, so I wasn't sure.  Thanks for the tip.